packages/core/src/oidc/utils.ts

```diff
@@ -8,3 +15,33 @@ export const buildOidcClientMetadata = (metadata?: OidcClientMetadata): OidcClie
   postLogoutRedirectUris: [],
   ...metadata,
 });
+
+const isOrigin = (value: string) => {
+  try {
+    const { origin } = new URL(value);
+
+    // Origin: <scheme> "://" <hostname> [ ":" <port> ]
+    return value === origin;
+  } catch {
+    return false;
+  }
+};
+
+export const validateCustomClientMetadata = (key: string, value: unknown) => {
+  const result = customClientMetadataGuard.pick({ [key]: true }).safeParse({ [key]: value });
+
+  if (!result.success) {
+    throw new errors.InvalidClientMetadata(key);
+  }
+
+  if (
+    key === CustomClientMetadataKey.corsAllowedOrigins &&
+    Array.isArray(value) &&
```

Comment:
technically isArray will be guarded by the previous customClientMetadataGuard validation already?

Reply:
it's just used to narrow down the type

Reply:
nit: `value?.some`

Reply:
<img width="762" alt="image" src="https://user-images.githubusercontent.com/10594507/162411143-f0177bbf-97b3-4439-a947-509efd3d0509.png">

@wangsijie cannot infer the correct type here.